src/daemon/screenlock.vala

```diff
@@ -0,0 +1,375 @@
+                        output += " -c " + file.get_path();
+                        break;
+                    }
+			    }
+
+                string[] style_configs = {
+                    @"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.css",
+                    @"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.css"
+                };
+
+                foreach (string? filepath in style_configs) {
+                    File file = File.new_for_uri(filepath);
+                    bool tmp = file.query_exists();
+                    if (tmp) {
+                        output += " -s " + file.get_path();
+                        break;
+                    }
+			    }
+            }
+
+            return output;
+        }
+
+        private string calculate_sleep() {
+            string output = "";
+
+            // number of seconds on ac power that is inactive before action taken; 0 is never
+            int sleep_inactive_ac_timeout = this.power.get_int("sleep-inactive-ac-timeout");
+
+            // BCC sets "nothing" or "suspend" when toggled
+            string sleep_inactive_ac_type = this.power.get_string("sleep-inactive-ac-type");
+
+            // number of seconds on battery that is inactive before action taken; 0 is never
+            int sleep_inactive_battery_timeout = this.power.get_int("sleep-inactive-battery-timeout");
+
+            // BCC sets "nothing" or "suspend" when toggled
+            string sleep_inactive_battery_type = this.power.get_string("sleep-inactive-battery-type");
+
+            if (sleep_inactive_battery_type == "suspend" && sleep_inactive_battery_timeout != 0) {
+                output = "timeout " + sleep_inactive_battery_timeout.to_string();
+                output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean true\" > /dev/null; then systemctl suspend; fi' ";
+            }
+
+            if (sleep_inactive_ac_type == "suspend" && sleep_inactive_ac_timeout !=0) {
+                output += " timeout " + sleep_inactive_ac_timeout.to_string();
+                output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean false\" > /dev/null; then systemctl suspend; fi' ";
+            }
+
+            return output;
+        }
+
+        private string calculate_dim() {
+            isdimmable = this.power.get_boolean("idle-dim");
+
+            if (!isdimmable) return "";
+
+            return "timeout 30 'dbus-send --type=method_call --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.Dim' resume 'dbus-send --type=method_call --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.Undim' ";
+        }
```

Comment:
Why do this when we have the methods available to us right in this class?

Reply:
'fraid I don't understand the comment.  Here we are adding a timeout/resume to swayidle to perform the dim/undim stuff if its set in BCC to screendim

Reply:
After going through this with Josh, I now understand that this is passed as an argument to `swayidle`. That bit wasn't clear to me, so I thought this was trying to call our own functions via dbus-send from our own binary.